save-demo/src/main/kotlin/com/saveourtool/save/demo/utils/KubernetesUtils.kt

```diff
@@ -0,0 +1,136 @@
+            ttlSecondsAfterFinished = TTL_AFTER_COMPLETED
+            backoffLimit = 0
+            template = PodTemplateSpec().apply {
+                spec = PodSpec().apply {
+                    if (kubernetesSettings.useGvisor) {
+                        nodeSelector = mapOf(
+                            "gvisor" to "enabled"
+                        )
+                        runtimeClassName = "gvisor"
+                    }
+                    metadata = ObjectMeta().apply {
+                        labels = mapOf(
+                            DEMO_ORG_NAME to demo.organizationName,
+                            DEMO_PROJ_NAME to demo.projectName,
+                            DEMO_VERSION to "manual",
+                            // "baseImageName" to baseImageName
+                            "io.kompose.service" to "save-demo-agent",
+                        )
+                    }
+                    // If agent fails, we should handle it manually (update statuses, attempt restart etc.)
+                    restartPolicy = "Never"
+                    containers = listOf(demoAgentContainerSpec(demo.sdk.toSdk().baseImageName(), kubernetesSettings))
+                }
+            }
+        }
+    }
+    logger.debug { "Attempt to create Job from the following spec: $job" }
+    try {
+        resource(job).create()
+        with(demo) {
+            logger.info("Created Job for demo $organizationName/$projectName")
+        }
+    } catch (kex: KubernetesClientException) {
+        with(demo) {
+            throw KubernetesRunnerException("Unable to create a job for demo $organizationName/$projectName", kex)
+        }
+    }
+}
+
+/**
+ * @param demo demo entity
+ * @param namespace namespace that the job should be associated with
+ * @return [ScalableResource] of [Job]
+ */
+fun KubernetesClient.getJob(demo: Demo, namespace: String): ScalableResource<Job> = batch()
+    .v1()
+    .jobs()
+    .inNamespace(namespace)
+    .withName(jobNameForDemo(demo))
+
+/**
+ * @param demo demo entity
+ * @param namespace namespace that the job should be associated with
+ * @return true if the resource is ready or exists (if no readiness check exists), false otherwise.
+ */
+fun KubernetesClient.isJobReady(demo: Demo, namespace: String) = getJob(demo, namespace).isReady
+
+/**
+ * @param demo demo entity
+ * @return list of ips of pods that are run by job associated with [demo]
+ */
+fun KubernetesClient.getJobPodsIps(demo: Demo) = getJobPods(demo)
+    .map { it.status.podIP }
+
+/**
+ * @param demo demo entity
+ * @return list of pods that are run by job associated with [demo]
+ */
+fun KubernetesClient.getJobPods(demo: Demo): List<Pod> = pods().withLabel("job-name", jobNameForDemo(demo))
+    .list()
+    .items
+
+/**
+ * @param demo demo entity
+ * @return name of job that is/should be assigned to [demo]
+ */
+fun jobNameForDemo(demo: Demo) = with(demo) { "demo-${organizationName.lowercase()}-${projectName.lowercase()}-1" }
```

Comment:
do you need `-1`? and do we need to have a version in jobName?

Reply:
Suggestion to postpone versioning until ph2